compaction/compaction_manager.hh

```diff
@@ -72,44 +72,88 @@ private:
             return compaction_disabled_counter > 0;
         }
     };
+
 public:
-    // TODO: turn task into a class
-    struct task {
-        compaction_manager& cm;
-        replica::table* compacting_table = nullptr;
-        shared_future<> compaction_done = make_ready_future<>();
-        exponential_backoff_retry compaction_retry = exponential_backoff_retry(std::chrono::seconds(5), std::chrono::seconds(300));
-        sstables::compaction_type type = sstables::compaction_type::Compaction;
-        bool compaction_running = false;
-        utils::UUID output_run_identifier;
-        sstables::compaction_data compaction_data;
-        compaction_state& compaction_state;
-        gate::holder gate_holder;
-
-        explicit task(compaction_manager& mgr, replica::table* t, sstables::compaction_type type)
-            : cm(mgr)
-            , compacting_table(t)
-            , type(type)
-            , compaction_state(cm.get_compaction_state(t))
-            , gate_holder(compaction_state.gate.hold())
+    class task {
+    protected:
+        compaction_manager& _cm;
+        replica::table* _compacting_table = nullptr;
```

Comment:
If this is never null in practice, it should be a reference (can be a follow-up).

Reply:
We use a pointer for historical reasons that required loose coupling.
I think that changing it to a reference is possible, but it's a bit out of scope for this series.
Let's consider doing it as a followup.

Reply:
Will defer for a followup patch/series.